src/rgw/rgw_rest_client.cc

```diff
@@ -813,6 +813,10 @@ int RGWRESTStreamRWRequest::do_send_prepare(RGWAccessKey *key, map<string, strin
                                          bufferlist *send_data)
 {
   string new_url = url;
+  if (new_url.empty() && host_style != VirtualStyle) {
+    ldout(cct, 5) << "ERROR: do_send_prepare has empty target url" << dendl;
+    return -EINVAL;
+  }
```

Comment:
i don't think we need a new error path, and this doesn't prevent the next line from crashing with VirtualStyle access

can we an empty check to that instead? something like:
```c++
if (!new_url.empty() && new_url.back() != '/')
```

Reply:
> i don't think we need a new error path, and this doesn't prevent the next line from crashing with VirtualStyle access
> 
> can we an empty check to that instead? something like:
> 
> ```c++
> if (!new_url.empty() && new_url.back() != '/')
> ```

THX. i have modifide the code